dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -346,73 +346,64 @@ protected void githubWebhookDelete(String repository, String gitReference, Strin
+            isSuccessful &= createWorkflowsAndVersionsFromDockstoreYml(dockstoreYaml12.getTools(), repository, gitReference, installationId, username, dockstoreYml, AppTool.class, messageWriter);
+
+        } catch (Exception ex) {
+
+            // If an exception propagates to here, log something helpful and abort .dockstore.yml processing.
+            isSuccessful = false;
+            String msg = "User " + username + ": Error handling push event for repository " + repository + " and reference " + gitReference + "\n" + generateMessageFromException(ex);
             LOG.info(msg, ex);
-            rollbackAndBeginNewTransaction();
-            LambdaEvent lambdaEvent = createBasicEvent(repository, gitReference, username, LambdaEvent.LambdaEventType.PUSH);
-            lambdaEvent.setSuccess(false);
-            lambdaEvent.setMessage(errorMessage);
-            lambdaEventDAO.create(lambdaEvent);
-            sessionFactory.getCurrentSession().getTransaction().commit();
+            messageWriter.println(msg);
+            messageWriter.println("Terminated processing of .dockstore.yml.");
+
             throw new CustomWebApplicationException(msg, statusCodeForLambda(ex));
-        } catch (Exception ex) {
-            endRateLimit = gitHubSourceCodeRepo.getGhRateLimitQuietly();
+
+        } finally {
+
+            // Make an entry in the github apps logs.
+            final boolean finalIsSuccessful = isSuccessful;
+            new TransactionHelper(sessionFactory).transaction(() -> {
+                LambdaEvent lambdaEvent = createBasicEvent(repository, gitReference, username, LambdaEvent.LambdaEventType.PUSH);
+                lambdaEvent.setSuccess(finalIsSuccessful);
+                setEventMessage(lambdaEvent, stringWriter.toString());
+                lambdaEventDAO.create(lambdaEvent);
+            });
+
+            GHRateLimit endRateLimit = gitHubSourceCodeRepo.getGhRateLimitQuietly();
             gitHubSourceCodeRepo.reportOnGitHubRelease(startRateLimit, endRateLimit, repository, username, gitReference, isSuccessful);
-            String msg = "User " + username + ": Unhandled error while handling push event for repository " + repository + " and reference " + gitReference + "\n" + ex.getMessage();
-            LOG.error(msg, ex);
-            rollbackAndBeginNewTransaction();
-            LambdaEvent lambdaEvent = createBasicEvent(repository, gitReference, username, LambdaEvent.LambdaEventType.PUSH);
-            lambdaEvent.setSuccess(false);
-            lambdaEvent.setMessage(ex.getMessage());
-            lambdaEventDAO.create(lambdaEvent);
-            sessionFactory.getCurrentSession().getTransaction().commit();
-            throw new CustomWebApplicationException(msg, LAMBDA_FAILURE);
+        }
+
+        if (!isSuccessful) {
+            throw new CustomWebApplicationException("At least one entry in .dockstore.yml could not be registered.", LAMBDA_FAILURE);
```

Comment:
Proposing "processed" instead of "registered" because the .dockstore.yml entry might already exist on Dockstore